libraries/ESP32/examples/Zigbee/Zigbee_Light_Switch/Zigbee_Light_Switch.ino

```diff
@@ -0,0 +1,317 @@
+// Copyright 2023 Espressif Systems (Shanghai) PTE LTD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @brief This example demonstrates simple Zigbee light switch.
+ * 
+ * The example demonstrates how to use ESP Zigbee stack to control a light bulb.
+ * The light bulb is a Zigbee end device, which is controlled by a Zigbee coordinator.
+ * Button switch and Zigbee runs in separate tasks.
+ * 
+ * Please check the README.md for more detailed description.
+ * 
+ */
+
+#ifndef ZIGBEE_MODE_ZCZR
+#error "Zigbee coordinator mode is not selected in Tools->Zigbee mode"
+#endif
+
+#include "esp_zigbee_core.h"
+#include "nvs_flash.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "ha/esp_zigbee_ha_standard.h"
+
+/* Switch configuration */
+#define GPIO_INPUT_IO_TOGGLE_SWITCH  GPIO_NUM_9
+#define PAIR_SIZE(TYPE_STR_PAIR) (sizeof(TYPE_STR_PAIR) / sizeof(TYPE_STR_PAIR[0]))
+
+typedef enum {
+    SWITCH_ON_CONTROL,
+    SWITCH_OFF_CONTROL,
+    SWITCH_ONOFF_TOGGLE_CONTROL,
+    SWITCH_LEVEL_UP_CONTROL,
+    SWITCH_LEVEL_DOWN_CONTROL,
+    SWITCH_LEVEL_CYCLE_CONTROL,
+    SWITCH_COLOR_CONTROL,
+} switch_func_t;
+
+typedef struct {
+    uint8_t pin;
+    switch_func_t func;
+} switch_func_pair_t;
+
+typedef enum {
+    SWITCH_IDLE,
+    SWITCH_PRESS_ARMED,
+    SWITCH_PRESS_DETECTED,
+    SWITCH_PRESSED,
+    SWITCH_RELEASE_DETECTED,
+} switch_state_t;
+
```

Comment:
Should all this typedef be part of a higher ZigBee Arduino API?
I think that we should thing more about creating a ZigBee Class, with Arduino APIs. Something that may be inspired in Arduino WiFi or BLE classes.